modeling/algorithm/shared-global-algorithm-exclude_test.go

```diff
@@ -0,0 +1,276 @@
+limitations under the License.
+*/
+
+package algorithm
+
+import (
+	"testing"
+
+	"github.com/googleinterns/k8s-topology-simulator/modeling/types"
+)
+
+func TestSharedGlobalAlgorithmExclude(t *testing.T) {
+	testCases := []testCase{
+		testCase{
+			name: "2 zones with no endpoints",
+			input: []types.Zone{
+				types.Zone{
+					Nodes:     30,
+					Endpoints: 100,
+					Name:      "ZoneA",
+				},
+				types.Zone{
+					Nodes:     30,
+					Endpoints: 0,
+					Name:      "ZoneB",
+				},
+				types.Zone{
+					Nodes:     30,
+					Endpoints: 0,
+					Name:      "ZoneC",
+				},
+			},
+			expectedOutput: map[string]types.EndpointSliceGroup{
+				"ZoneA": types.EndpointSliceGroup{
+					Label: "ZoneA",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 0, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneA": 1,
+					},
+				},
+				"ZoneB": types.EndpointSliceGroup{
+					Label: "ZoneB",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 0, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneB": 1,
+					},
+				},
+				"ZoneC": types.EndpointSliceGroup{
+					Label: "ZoneC",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 0, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneC": 1,
+					},
+				},
+				"global": types.EndpointSliceGroup{
+					Label: "global",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 100, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneA": 1,
+						"ZoneB": 1,
+						"ZoneC": 1,
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		testCase{
+			name: "only 1 endpoint",
+			input: []types.Zone{
+				types.Zone{
+					Nodes:     30,
+					Endpoints: 1,
+					Name:      "ZoneA",
+				},
+				types.Zone{
+					Nodes:     30,
+					Endpoints: 0,
+					Name:      "ZoneB",
+				},
+				types.Zone{
+					Nodes:     30,
+					Endpoints: 0,
+					Name:      "ZoneC",
+				},
+			},
+			expectedOutput: map[string]types.EndpointSliceGroup{
+				"ZoneA": types.EndpointSliceGroup{
+					Label: "ZoneA",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 0, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneA": 1,
+					},
+				},
+				"ZoneB": types.EndpointSliceGroup{
+					Label: "ZoneB",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 0, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneB": 1,
+					},
+				},
+				"ZoneC": types.EndpointSliceGroup{
+					Label: "ZoneC",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 0, Weight: 1},
```

Comment:
Does `Number: 0` mean there are 0 endpoints in this slice group? If so, why would it exist?

Reply:
deleted